src/app/query.rs

```diff
@@ -1,449 +1,434 @@
+
 const DELIMITER_LIST: [char; 6] = ['=', '>', '<', '(', ')', '\"'];
 const COMPARISON_LIST: [&str; 3] = [">", "=", "<"];
 const OR_LIST: [&str; 2] = ["or", "||"];
 const AND_LIST: [&str; 2] = ["and", "&&"];
 
-/// I only separated this as otherwise, the states.rs file gets huge... and this should
-/// belong in another file anyways, IMO.
-pub trait ProcessQuery {
-    /// In charge of parsing the given query.
-    /// We are defining the following language for a query (case-insensitive prefixes):
-    ///
-    /// - Process names: No prefix required, can use regex, match word, or case.
-    ///   Enclosing anything, including prefixes, in quotes, means we treat it as an entire process
-    ///   rather than a prefix.
-    /// - PIDs: Use prefix `pid`, can use regex or match word (case is irrelevant).
-    /// - CPU: Use prefix `cpu`, cannot use r/m/c (regex, match word, case).  Can compare.
-    /// - MEM: Use prefix `mem`, cannot use r/m/c.  Can compare.
-    /// - STATE: Use prefix `state`, can use regex, match word, or case.
-    /// - USER: Use prefix `user`, can use regex, match word, or case.
-    /// - Read/s: Use prefix `r`.  Can compare.
-    /// - Write/s: Use prefix `w`.  Can compare.
-    /// - Total read: Use prefix `read`.  Can compare.
-    /// - Total write: Use prefix `write`.  Can compare.
-    ///
-    /// For queries, whitespaces are our delimiters.  We will merge together any adjacent non-prefixed
-    /// or quoted elements after splitting to treat as process names.
-    /// Furthermore, we want to support boolean joiners like AND and OR, and brackets.
-    fn parse_query(&self) -> Result<Query>;
-}
+/// In charge of parsing the given query.
+/// We are defining the following language for a query (case-insensitive prefixes):
+///
+/// - Process names: No prefix required, can use regex, match word, or case.
+///   Enclosing anything, including prefixes, in quotes, means we treat it as an entire process
+///   rather than a prefix.
+/// - PIDs: Use prefix `pid`, can use regex or match word (case is irrelevant).
+/// - CPU: Use prefix `cpu`, cannot use r/m/c (regex, match word, case).  Can compare.
+/// - MEM: Use prefix `mem`, cannot use r/m/c.  Can compare.
+/// - STATE: Use prefix `state`, can use regex, match word, or case.
+/// - USER: Use prefix `user`, can use regex, match word, or case.
+/// - Read/s: Use prefix `r`.  Can compare.
+/// - Write/s: Use prefix `w`.  Can compare.
+/// - Total read: Use prefix `read`.  Can compare.
+/// - Total write: Use prefix `write`.  Can compare.
+///
+/// For queries, whitespaces are our delimiters.  We will merge together any adjacent non-prefixed
+/// or quoted elements after splitting to treat as process names.
+/// Furthermore, we want to support boolean joiners like AND and OR, and brackets.
+pub fn parse_query(
+    search_query: &str, is_searching_whole_word: bool, is_ignoring_case: bool,
+    is_searching_with_regex: bool,
+) -> Result<Query> {
```

Comment:
99% of these changes copy-pasted from tuine branch. In the future I'm planning on entirely rewriting this though, will add tests then.